lesson15/main.js

```diff
@@ -86,41 +88,49 @@ const displayModal= () => {
   overlay.style.display= "block";
 };
 
+const resetInput = () => {
+  const inputFields = document.getElementsByClassName("js-input");
+
+  for (errorMessage of errorMessages) {
+    errorMessage.textContent =""
+  };
+  for(inputField of inputFields) {
+    inputField.value = ""
+    inputField.style = "border border-slate-600 "
+  };
+}
+
 const closeModal = () => {
-  resetInput()
   modal.style.display="none";
   overlay.style.display="none";
+  resetInput();
 };
 
 const removeModal = () =>  document.getElementById("js-modal-wrapper").remove();
 
-const resetInput = () => {
-  numberInput.value = '';
-  validationMessage.style.display ="none";
-};
-
-const displayValidation = (message) => {
-  validationMessage.textContent = `${message}`;
-  validationMessage.style.display="block";
-}
-
-openListButton.addEventListener("click", () => {
-  const userInput = numberInput.value;
-  if(userInput === ''){
-    displayValidation("Input feild cannot be empty");
-    return;
-  }
-  if (!userInput.match(/^\d+$/)){
-    displayValidation("Input value must be a number");
-    return;
+const checkValidation = (id, serial, message) => {
+  if(id.value === "") {
+    errorMessages[serial].textContent = message;
+    id.style.border = "2px red solid";
+    id.style.margin = "0px";
+  } else {
+    errorMessages[serial].textContent = "";
+    id.style = "border border-slate-600";
   }
-  removeModal();
-  displayList(userInput);
-});
+};
 
-numberInput.addEventListener("click", ()=>{
-  resetInput();
-})
 openButton.addEventListener("click", displayModal);
 closeButton.addEventListener("click", closeModal);
 overlay.addEventListener("click", closeModal);
+
+form.addEventListener("submit", (e) => { 
+  const userNumberInput = document.getElementById("js-number-input");
+  const userNameInput = document.getElementById("js-name-input");
+  if(userNameInput.value !== "" && userNumberInput.value !== ""){
+    removeModal();
+    displayList(userNameInput.value, userNumberInput.value);
+  }
+  e.preventDefault();
+  checkValidation(userNameInput, 0, "Name cannot be blank");
+  checkValidation(userNumberInput, 1, "Number cannot be blank");
+});
```

Comment:
IMO
It is possible to SUBMIT even if there is only a space in the name field. I think this can be solved by using trim().

Reply:
I appreciate the comment as I did not realise that user can submit when the input value is only space(s). Thank you for the suggestion to fix it. 
Please see the update.
6ad544c96b5c4e21ce5ff7a81fa84174f72e7142